aws/resource_aws_gamelift_game_session_queue.go

```diff
@@ -0,0 +1,204 @@
+	out, err := conn.CreateGameSessionQueue(&input)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*out.GameSessionQueue.GameSessionQueueArn)
+	d.Set("name", out.GameSessionQueue.Name)
+
+	return resourceAwsGameliftGameSessionQueueRead(d, meta)
+}
+
+func resourceAwsGameliftGameSessionQueueRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).gameliftconn
+	log.Printf("[INFO] Describing Gamelift Session Queues: %s", d.Get("name"))
+	limit := int64(1)
+	out, err := conn.DescribeGameSessionQueues(&gamelift.DescribeGameSessionQueuesInput{
+		Names: aws.StringSlice([]string{d.Get("name").(string)}),
+		Limit: &limit,
+	})
+	if err != nil {
+		if isAWSErr(err, gamelift.ErrCodeNotFoundException, "") {
+			log.Printf("[WARN] Gamelift Session Queues (%s) not found, removing from state", d.Get("name"))
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+	sessionQueues := out.GameSessionQueues
+
+	if len(sessionQueues) < 1 {
+		log.Printf("[WARN] Gamelift Session Queue (%s) not found, removing from state", d.Get("name"))
+		d.SetId("")
+		return nil
+	}
+	if len(sessionQueues) != 1 {
+		return fmt.Errorf("expected exactly 1 Gamelift Session Queues, found %d under %q",
+			len(sessionQueues), d.Get("name"))
+	}
+	sessionQueue := sessionQueues[0]
+
+	d.Set("arn", sessionQueue.GameSessionQueueArn)
+	d.Set("name", sessionQueue.Name)
+	d.Set("timeout_in_seconds", sessionQueue.TimeoutInSeconds)
+	d.Set("destinations", sessionQueue.Destinations)
+	if err := d.Set("player_latency_policy", flattenGameliftPlayerLatencyPolicies(sessionQueue.PlayerLatencyPolicies)); err != nil {
+		return fmt.Errorf("error setting player_latency_policy: %s", err)
+	}
+
+	return nil
+}
+
+func flattenGameliftPlayerLatencyPolicies(playerLatencyPolicies []*gamelift.PlayerLatencyPolicy) []interface{} {
+	lst := []interface{}{}
+	for _, policy := range playerLatencyPolicies {
+		m := map[string]interface{}{
+			"maximum_individual_player_latency_milliseconds": aws.Int64Value(policy.MaximumIndividualPlayerLatencyMilliseconds),
+			"policy_duration_seconds":                        aws.Int64Value(policy.PolicyDurationSeconds),
+		}
+		lst = append(lst, m)
+	}
+	return lst
+}
+
+func resourceAwsGameliftGameSessionQueueUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).gameliftconn
+
+	name := d.Get("name").(string)
+
+	log.Printf("[INFO] Updating Gamelift Session Queue: %s", name)
+
+	input := gamelift.UpdateGameSessionQueueInput{
+		Name:                  aws.String(d.Get("name").(string)),
+		Destinations:          expandGameliftGameSessionQueueDestinations(d.Get("destinations").([]interface{})),
+		PlayerLatencyPolicies: expandGameliftGameSessionPlayerLatencyPolicies(d.Get("player_latency_policy").([]interface{})),
+		TimeoutInSeconds:      aws.Int64(int64(d.Get("timeout_in_seconds").(int))),
+	}
+
+	_, err := conn.UpdateGameSessionQueue(&input)
+	if err != nil {
+		return err
```

Comment:
Nit: We should return context about when errors occur for operators and code maintainers, e.g.

```suggestion
		return fmt.Errorf("error updating Gamelift Game Session Queue (%s): %s", d.Id(), err)
```

Reply:
:+1: